### ISSUE TITLE
Error catching `subprocess.CalledProcessError`

### ISSUE THREAD

**Issue by CJ-Wright** [xonsh/xonsh#2772]
It seems that try except isn't properly catching `subprocess.CalledProcessError`

## xonfig
```
+------------------+---------------------+
| xonsh            | 0.6.7.dev4830       |
| Git SHA          | bdda27f             |
| Commit Date      | Jul 3 10:58:14 2018 |
| Python           | 3.6.3               |
| PLY              | 3.9                 |
| have readline    | True                |
| prompt toolkit   | 1.0.15              |
| shell type       | prompt_toolkit      |
| pygments         | 2.2.0               |
| on posix         | True                |
| on linux         | True                |
| distro           | debian              |
| on darwin        | False               |
| on windows       | False               |
| on cygwin        | False               |
| on msys2         | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
+------------------+---------------------+
```
## Expected Behavior
```python
from xonsh.lib import subprocess
try:
    subprocess.check_call(['false'])
except subprocess.CalledProcessError:
    print('hi')
    return
```
Should print, not error.

## Current Behavior
It seems that xonsh escapes and just sends the error out and halts, giving the following error.
```python
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
subprocess.CalledProcessError: Command '['/bin/false']' returned non-zero exit status 1.
```
## Steps to Reproduce
see above

**Comment by scopatz** [comment]
I don't think this is about try-except per se, but rather the implementation of `check_call()` and friends

**Comment by scopatz** [comment]
Potential fix in #2797
